package-res/lib/protovis.js

```diff
@@ -3670,9 +3670,11 @@ pv.Scale.quantitative = function() {
 
   /** @private */
   function scale(x) {
+    if (x >= Math.pow(2,53)) { x = x / Math.pow(2,53); }
     var j = pv.search(d, x);
     if (j < 0) j = -j - 2;
     j = Math.max(0, Math.min(i.length - 1, j));
+    if(j+1>=l.length) { j = j-1; }
```

Comment:
By what I could understand from the code, it is always true that `d.length === l.length === r.length === i.length - 1`.
Because of the code `j = Math.max(0, Math.min(i.length - 1, j));`, the index `j` is always be a valid index of the array `i`. Thus, it must be a valid index of all of the other three arrays as well (they all have one more position).
The condition `j >= l.length - 1` will never be true cause `j <= i.length - 1`, which is equivalent to, `j <= l.length - 2`.
Thus, if I'm not mistaken, the added line has no effect.